libs/designsystem/calendar/src/calendar.component.ts

```diff
@@ -304,18 +302,17 @@ export class CalendarComponent implements OnInit, AfterViewInit, OnChanges {
     return startOfDay(dateLocalOrUTC);
   }
 
-  private getWeekDays(): string[] {
+  private getWeekDays(): WeekDay[] {
     const now = new Date();
     const week = eachDayOfInterval({
       start: startOfWeek(now, { locale: this.locale }),
       end: endOfWeek(now, { locale: this.locale }),
     });
 
-    return week.map((date) => this.getFirstLetterOfWeekDayCapitalized(date));
-  }
-
-  private getFirstLetterOfWeekDayCapitalized(date: Date) {
-    return this.formatWithLocale(date, 'EEEEE');
+    return week.map((date) => ({
+      firstLetterCapitalized: this.formatWithLocale(date, 'EEEEE'), // EEEEE returns the first letter of the day capitalized
+      fullName: this.formatWithLocale(date, 'EEEE'), // EEEE returns the full name of the day
```

Comment:
Consider naming the formats instead of using comments:
```suggestion
      const weekdayNarrowFormat = 'EEEEE';
      const weekdayWideFormat = 'EEEE';
      firstLetterCapitalized: this.formatWithLocale(date, weekdayNarrowFormat),
      fullName: this.formatWithLocale(date, weekdayWideFormat),
```